common/src/main/scala/com/gu/localnews/common/services/index/CouncilPetitionsIndex.scala

```diff
@@ -0,0 +1,80 @@
+package com.gu.localnews.common.services.index
+
+import com.sksamuel.elastic4s.http.ElasticDsl._
+import com.gu.localnews.common.CouncilPetition
+import com.sksamuel.elastic4s.http.ElasticClient
+import com.sksamuel.elastic4s.http.index.CreateIndexResponse
+import com.sksamuel.elastic4s.http.index.admin.IndexExistsResponse
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+
+trait CouncilPetitionsIndex {
+  def client: ElasticClient
+
+  def setupCouncilPetitions()(implicit ec: ExecutionContext) = {
+    Await.ready(client.execute(indexExists("council-petitions")).flatMap { resp =>
+      println("Checking if we need to create council petitions index...")
+      resp.result match {
+        case IndexExistsResponse(false) =>
+
+          println("Index did not already exist - creating")
+          client.execute(createIndex("council-petitions").mappings(
+            mapping("_doc").as(
+              textField("title"),
+              textField("petitionLink"),
+              textField("description"),
+              textField("backgroundInfo"),
+              textField("closingDate"), // Make date field?
+              textField("creator"),
+              textField("signatures"),
+              objectField("entities").fields(
+                textField("people"),
+                textField("places"),
+                textField("organisations"),
+                textField("dates"), // TOOD make dates fields?
+                textField("keyPhrases"),
+                textField("sentiment"),
+              )
+            )
+          ))
+        case _ =>
+          println("Index already exists!")
+          Future.successful()
+      }
+    }, 10 seconds)
+  }
+
+
+  def insertCouncilPetitions(petition: CouncilPetition)(implicit ec: ExecutionContext): Unit = {
+    val values = Map(
+      "title" -> petition.title,
+      "petitionLink" -> petition.petitionLink,
+      "description" -> petition.description,
+      "backgroundInfo" -> petition.backgroundInfo,
+      "closingDate" -> petition.closingDate, // Make date field?
+      "creator" -> petition.creator,
+      "signatures" -> petition.signatures,
+      "entities" -> Map(
+        "people" -> petition.entities.people,
+        "places" -> petition.entities.places,
+        "organisations" -> petition.entities.organisations,
+        "dates" -> petition.entities.dates, // TOOD make dates fields?
+        "keyPhrases" -> petition.entities.keyPhrases,
+        "sentiment" -> petition.entities.sentiment,
+      )
+    ) ++
+      petition.description.map("description" -> _) ++
+      petition.backgroundInfo.map("backgroundInfo" -> _) ++
+      petition.signatures.map("signatures" -> _)
```

Comment:
@itsibitzi this is coming out like:
```
"signatures": [
"PetitionSignature(Jordana  Yechiel)",
"PetitionSignature(Julia  Torr)",
"PetitionSignature(Pedro  Campos)",
"PetitionSignature(Andy  Comte-Lenestour)"
```
But it won't let me do `_.name` whyyyyyyy?